src/mandelbrot.rs

```diff
@@ -0,0 +1,104 @@
+// The Computer Language Benchmarks Game
+// http://benchmarksgame.alioth.debian.org/
+//
+// contributed by Matt Watson
+// contributed by TeXitoi
+
+use std::io::Write;
+use std::thread;
+const THREADS: usize = 8;
+const MAX_ITER: usize = 50;
+const VLEN: usize = 8;
+const ZEROS: Vecf64 = [0.0; VLEN];
+
+pub type Vecf64 = [f64; VLEN];
+
+fn mul2 (x: Vecf64, y: Vecf64) -> Vecf64 {
+    let mut res = ZEROS;
+    for i in 0..VLEN {res[i] = x[i] * y[i];}
+    res
+}
+fn add2 (x: Vecf64, y: Vecf64) -> Vecf64 {
+    let mut res = ZEROS;
+    for i in 0..VLEN {res[i] = x[i] + y[i];}
+    res
+}
+fn sub2 (x: Vecf64, y: Vecf64) -> Vecf64 {
+    let mut res = ZEROS;
+    for i in 0..VLEN { res[i] = x[i] - y[i]; }
+    res
+}
+
+pub fn mbrot8(cr: Vecf64, ci: Vecf64) -> u8 {
```

Comment:
8 is sub-optimal from a vectorisation standpoint, but it fitted so neatly with the whole pbm thing that it reduced the code.

Strangely enabling AVX when testing on my system (not supported by the Q6600) did very little (5-10%), looks like it might be waiting for cache too often?